homeassistant/components/islamic_prayer_times/config_flow.py

```diff
@@ -49,13 +76,39 @@ async def async_step_user(
         self, user_input: dict[str, Any] | None = None
     ) -> FlowResult:
         """Handle a flow initialized by the user."""
-        if self._async_current_entries():
-            return self.async_abort(reason="single_instance_allowed")
+        errors = {}
 
-        if user_input is None:
-            return self.async_show_form(step_id="user")
+        if user_input is not None:
+            lat: float = user_input[CONF_LOCATION][CONF_LATITUDE]
+            lon: float = user_input[CONF_LOCATION][CONF_LONGITUDE]
+            await self.async_set_unique_id(f"{lat}-{lon}")
+            self._abort_if_unique_id_configured()
+
+            if not (errors := await async_validate_location(self.hass, lat, lon)):
+                return self.async_create_entry(
+                    title=user_input[CONF_NAME],
+                    data={
+                        CONF_LATITUDE: lat,
+                        CONF_LONGITUDE: lon,
+                    },
+                )
 
-        return self.async_create_entry(title=NAME, data=user_input)
+        home_location = {
+            CONF_LATITUDE: self.hass.config.latitude,
+            CONF_LONGITUDE: self.hass.config.longitude,
+        }
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(CONF_NAME, default=NAME): TextSelector(),
+                    vol.Required(
+                        CONF_LOCATION, default=home_location
```

Comment:
I _think_ the default is already the home location, can you double check?

Reply:
I tried removing the default value but then the config flow no longer opens. I am not sure if the location selector tries to auto detect the location or uses the stored home location. I prefer to make the default as the stored home location.

Reply:
That's fine by me!